### PR TITLE
Add Ada implementation

### PR DESCRIPTION
**Pull request by johnperry-math**
I added an example for the Ada programming language. Runtime is roughly that of c++ on my machine: only a very little slower.

### CODE REVIEW

File: ada/complete_unscientific.adb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  t: Tree.Tree;
+  cur: Integer := 5;
+  res: Integer := 0;
+  mode: Integer;
+  procedure PutI(item: Integer; width: Natural := 0; base: Natural := 10)
+    renames Ada.Integer_Text_IO.Put;
+  procedure New_Line(spacing: Ada.Text_IO.Positive_Count := 1) renames Ada.Text_IO.New_Line;
+begin
+  initialize;
+  for i in 1..999999 loop
```
Comment (frol) [inline ada/complete_unscientific.adb]: Is this range inclusive?

Comment (johnperry-math) [reply ada/complete_unscientific.adb]: Yes. For example, the loop

    for i in 1..10 loop Put(i, 0); New_Line(1); end loop;

prints all the numbers from 1 to 10. I inferred from your code's starting at 1 and testing `i < 1000000` that your code looped 999999 times.

I also checked that the output was the same. The C++, Java, Nim, and Ada versions all report 331665. I actually had a different value at first on account of a small bug.

**Comment by frol** [comment]
I have just added Ada to the table (Linux only at the moment)! It is fast!

**Comment by johnperry-math** [comment]
It's safe, too! Catches all sorts of errors that C++ won't. (Including one that burned me a few months ago...)

**Pull request merged**